### ISSUE TITLE
Confusion between ColumnAttribute and ColumnNameAttribute should be prevented

### ISSUE THREAD

**Issue by jsboige** [dotnet/machinelearning#1603]
I just spent nearly a whole hour in front of a class full of students trying to figure out what was wrong with my code. I was demonstrating the SentimentAnalysis classifier and had the working code in split-screen to compare. My faulty code had a Column Attribute instead of ColumnName Attribute in the prediction class.
First, not a single pair of eyes noticed the error. For some reason, one does not seem to register the difference, however obvious it should be.
Then, although the subsequent exception did mention something was wrong with columns it was virtually impossible to debug:
ColumnAttribute actually has the corresponding "name" member, it's just that the constructor with the same signature sets the "ordinal" member instead, and that without failing unfortunately.
Accordingly, trying to compare the code execution step by step with complete source code, the only visible difference appears pretty late in SchemaDefinition.Create(..), when both attributes are tested for existence and the corresponding "name" value, with one of the two missing.
Then the resulting exception is triggered again much later in the TypedCursorable's private constructor, where the faulty column is considered missing.
This is deadly. 
If you don't notice the spelling difference visually right away, and again a whole class didn't for some reason, there is very little chance even debugging the whole engine will get you anywhere.

In short, it should be made impossible or near to impossible to use [Column("name")] instead of [ColumnName("name")] by mistake, and at least easier to diagnose.

**Comment by Zruty0** [comment]
@jsboige I'm sorry about that frustration. The new ML.NET API will at least produce earlier warnings for schema mismatches (so you would be notified that the column renaming didn't actually happen). 

In addition, `Column` attribute is really specific to `TextLoader`, so we will remove it as part of removing the legacy API. 

Right now, in the new API it is impossible to specify the schema of a text file by defining a class. We have #1515 to re-enable easy data reading in the new API.

**Comment by jsboige** [comment]
Thanks for letting me know and making sure this will prevented in the future.
